### PR TITLE
Jqian/detectsdk

### PR DESCRIPTION
**Pull request by jqian2017**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,25 +4,103 @@
...
+      - run:  # Access to @heroku/salesforce-sdk
           name: Middleware Build and Test Prep
           command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/.npmrc
-
       - run: npm install
-
       - run: npm run build
+      - run: npm run test
+      - run:
+          #work around to carray env variable between bash shells
```
Comment (cwallsfdc) [inline .circleci/config.yml]: "carry".

File: .circleci/config.yml [hunk 2]
```diff
@@ -4,25 +4,103 @@
...
+              fi
+      #save the packaged tgz for release job to use
+      - run: |
+          cd ~/repo
+          make package
+          ls -alh sf-fx-middleware-buildpack-$VERSION_TOML.tgz
+      - persist_to_workspace:
+          root: ~/repo
+          paths:
+            - sf-fx-middleware-buildpack-*.tgz
```
Comment (cwallsfdc) [inline .circleci/config.yml]: Will this commit to repo?

Comment (cwallsfdc) [reply .circleci/config.yml]: Nvm, I see below.

File: ci-git-release.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+if [ "$CIRCLE_BRANCH" = "jqian/detectsdk" ]; then
```
Comment (cwallsfdc) [inline ci-git-release.sh]: Intent to have "jqian/detectsdk"?

File: ci-git-release.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -eo pipefail
```
Comment (cwallsfdc) [inline ci-git-release.sh]: Add comment on what script does.

Comment (cwallsfdc) [reply ci-git-release.sh]: How/when is this script used?

Comment (jqian2017) [reply ci-git-release.sh]: I no longer use this script... as all the logic is now part of the circleci config.yml

I could delete it

File: ci-git-release.sh (hunk 3 above)
Comment (jqian2017) [reply ci-git-release.sh]: I no longer use this script... as all the logic is now part of the circleci config.yml... I will delete the script file

**Pull request merged**